### ISSUE TITLE
How to integrate mvnd in jenkins?

### ISSUE THREAD

**Issue by hongchangliu** [apache/maven-mvnd#592]
How to integrate mvnd in jenkins?which plugin？

**Comment by gnodet** [comment]
Why do you want to do tha

**Comment by hongchangliu** [comment]
@gnodet We are now using Jenkins' maven command to build code, how to build through mvnd

**Comment by hongchangliu** [comment]
![image](https://user-images.githubusercontent.com/24295943/151477473-f9a0ac8b-196c-4406-a8a3-daa101e714d3.png)

**Comment by ppalaga** [comment]
> @gnodet We are now using Jenkins' maven command to build code, how to build through mvnd

@hongchangliu you said what you do and what you'd like to do, but you have not answered @gnodet's question, why do you want to do that? What expectations do you have from mvnd on Jenkins? By answering those questions, you'd perhaps realize yourself that your expectations are unrealistic. But maybe not, it depends on what you really expect.

**Comment by hongchangliu** [comment]
Now jenkins has a plugin for maven, I want to know if there is a plugin for maven-mvnd, which can be easily configured and built with maven-mvnd

**Comment by hongchangliu** [comment]
@gnodet @ppalaga Now jenkins has a plugin for maven, I want to know if there is a plugin for maven-mvnd, which can be easily configured and built with maven-mvnd?

**Comment by gnodet** [comment]
> @gnodet @ppalaga Now jenkins has a plugin for maven, I want to know if there is a plugin for maven-mvnd, which can be easily configured and built with maven-mvnd?

`mvnd` is a command line tool that aims at easing / speeding developers performing builds on the command line.  It's currently not integrated in CI builds.

**Comment by TOP-LH** [comment]
> @gnodet@ppalaga现在 jenkins 有一个 maven 插件，我想知道是否有 maven-mvnd 的插件，可以使用 maven-mvnd 轻松配置和构建？

你可以使用jenkinsfile(流水线工程)去让jenkins执行mvnd的命令

**Comment by marioja** [comment]
@gnodet Guillaume, even though mvnd is a command line tools and even though it is not currently integrated in CI builds (which is why @hongchangliu open this ticket) why are you summarily stating that because it is not currently integrated it should never be and that any request for such integration should be immediately closed?  I do not know the history of mvnd and if you are a contributor and if you are, I thank you for all your hard work.  However, I think closing an issue this way is rather impolite at best.  I was going to ask the exact same question as I would like to have my CI build benefit from the speed improvements mvnd provides when my CI builds run.

**Comment by ppalaga** [comment]
Using mvnd on a CI may or may not be a good idea depending on what you expect. 

* Keeping a single daemon process up and running on CI worker nodes so that it can be used for multiple non-related builds might not be a good idea: it would perhaps be hard to implement and it might harm the reproducibility of the builds. mvnd is still relatively new and thus potentially buggy.
* Using mvnd for multiple subsequent builds within the same CI job on the same worker node (such as first `mvnd install -DskipTests` in parallel and then `cd integration-tests && mvnd test` with or without `-1/--serial`) might work. Just make sure that the daemon process does not survive across your CI job boundaries. In this way you may benefit from better parallel builds (better logging and better CPU saturation compared to stock Maven) and from keeping the the daemon JVM warm which allows for faster execution on subsequent mvnd invocations.

Given the above, feel free to try what is the best for you. 

I tend to agree with @gnodet that implementing or advising about any given CI's configuration is out of scope here. You may want to contact your CI vendor for that.
